File: 908-middle-of-the-linked-list/middle-of-the-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {number}
 */
var listLength = function(head) {
    let length = 0;
    let current = head;
    while (current !== null) {
        length++;
        current = current.next;
    }
    return length;
};

/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    const length = listLength(head);
    const middle = Math.floor(length / 2);
    let temp = head;
    let count = 0;
    while (count < middle) {
        temp = temp.next;
        count++;
    }
    return temp;
};

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.kt
/**
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */

class Solution {
    fun listLength(head: ListNode?): Int {
        var length = 0
        var current = head
        while (current != null) {
            length++
            current = current.next
        }
        return length
    }

    fun middleNode(head: ListNode?): ListNode? {
        val length = listLength(head)
        val middle = length / 2
        var temp = head
        var count = 0
        while (count < middle) {
            temp = temp?.next
            count++
        }
        return temp
    }
}